src/NetMQ/ReceivingSocketExtensions.cs

```diff
@@ -437,24 +1358,124 @@ public static bool WaitForSignal(this IReceivingSocket socket)
+        /// Attempt to receive a single frame from <see cref="socket"/>, then ignore its content.
+        /// If no message is immediately available, return <c>false</c>.
+        /// Indicate whether further frames exist via <paramref name="more"/>.
+        /// </summary>
+        /// <param name="socket">The socket to receive from.</param>
+        /// <param name="more"><c>true</c> if another frame of the same message follows, otherwise <c>false</c>.</param>
+        /// <returns><c>true</c> if a frame was received and ignored, otherwise <c>false</c>.</returns>
+        public static bool TrySkipFrame([NotNull] this IReceivingSocket socket, out bool more)
+        {
+            var msg = new Msg();
+            msg.InitEmpty();
+            var result = socket.TryReceive(ref msg, TimeSpan.Zero);
+            more = msg.HasMore;
+            msg.Close();
+            return result;
+        }
+
+        #endregion
+
+        #region Timeout
+
+        /// <summary>
+        /// Attempt to receive a single frame from <see cref="socket"/>, then ignore its content.
+        /// If no message is available within <paramref name="timeout"/>, return <c>false</c>.
+        /// </summary>
+        /// <param name="socket">The socket to receive from.</param>
+        /// <param name="timeout">The maximum period of time to wait for a message to become available.</param>
+        /// <returns><c>true</c> if a frame was received and ignored, otherwise <c>false</c>.</returns>
+        public static bool TrySkipFrame([NotNull] this IReceivingSocket socket, TimeSpan timeout)
+        {
+            var msg = new Msg();
+            msg.InitEmpty();
+            var received = socket.TryReceive(ref msg, timeout);
+            msg.Close();
+            return received;
+        }
+
+        /// <summary>
+        /// Attempt to receive a single frame from <see cref="socket"/>, then ignore its content.
+        /// If no message is available within <paramref name="timeout"/>, return <c>false</c>.
+        /// Indicate whether further frames exist via <paramref name="more"/>.
+        /// </summary>
+        /// <param name="socket">The socket to receive from.</param>
+        /// <param name="timeout">The maximum period of time to wait for a message to become available.</param>
+        /// <param name="more"><c>true</c> if another frame of the same message follows, otherwise <c>false</c>.</param>
+        /// <returns><c>true</c> if a frame was received and ignored, otherwise <c>false</c>.</returns>
+        public static bool TrySkipFrame([NotNull] this IReceivingSocket socket, TimeSpan timeout, out bool more)
+        {
+            var msg = new Msg();
+            msg.InitEmpty();
+
+            if (!socket.TryReceive(ref msg, timeout))
+            {
+                more = false;
+                msg.Close();
+                return false;
+            }
+
+            more = msg.HasMore;
+            msg.Close();
+            return true;
```

Comment:
another idea if you did skip frame is to do DropAll which drop the rest of the frames (until more). It is actually very useful because when you implement a protocol and there is an error you want to drop the entire frames in the message and continue to next message